app/settings/data-import/data-import.directive.js

```diff
@@ -1,42 +1,302 @@
+                                        key: item.key + '.lat',
+                                        label: item.label + ' (Latitude)',
+                                        priority: item.priority,
+                                        required: item.required
+                                    }, {
+                                        key: item.key + '.lon',
+                                        label: item.label + ' (Longitude)',
+                                        priority: item.priority,
+                                        required: item.required
+                                    }
+                                );
+                            }, [])
+                            .value();
+
+                        $scope.selectedForm.attributes  = _.chain($scope.selectedForm.attributes)
+                            .reject({type : 'point'})
+                            .concat(points)
+                            // Add in the Post specific mappable fields
+                            .push({
+                                    'key': 'title',
+                                    'label': $translate.instant('post.modify.form.title'),
+                                    'priority': 0,
+                                    'required': true
+                                },
+                                {
+                                    'key': 'content',
+                                    'label': $translate.instant('post.modify.form.description'),
+                                    'priority': 1,
+                                    'required': true
+                                },
+                                {
+                                    'key': 'tags',
+                                    'label': $translate.instant('post.modify.form.categories'),
+                                    'priority': 2
+                                }
+                            )
+                            .sortBy('priority')
+                            .value();
+
+                        _.each($scope.selectedForm.attributes, function (attribute, index) {
+                            if (attribute.required) {
+                                $scope.required_fields.push(attribute.key);
+                                $scope.required_fields_map[attribute.key] = attribute.label;
+                            }
+                        });
+
+                        _.each($scope.selectedForm.tasks, function (task) {
+                            if (task.required) {
+                                $scope.hasRequiredTask = true;
+                            }
+                        });
+                    });
                 }, function (errorResponse) {
                     Notify.apiErrors(errorResponse);
                 });
             };
+
+            // Check for missing required fields and return the missing fields
+            $scope.checkRequiredFields = function (fields) {
```

Comment:
Does this really need to be on `$scope`? it seems like its called from elsewhere in the directive code